syncrepo.sh

```diff
@@ -219,57 +225,57 @@ syncrepo.save_config() {
 
 # Debug wrapper to trace function calls
 utils.call() {
+    [[ $# -gt 0 ]] || return 1
     local name=$1 && shift
-    utils.say -d 'entering %s' "$name"
+    call_count=${call_count:-2}
+    utils.say -d '%-*s enter %s' $((call_count++)) '->' "$name"
     "$name" "$@"
-    utils.say -d 'leaving %s' "$name"
+    utils.say -d '%-*s leave %s' $((--call_count)) '<-' "$name"
 }
 
 # Log message and print to stdout
 # shellcheck disable=SC2059
 utils.say() {
```

Comment:
TODO: Modifiy this to write to stderr, where apropriate (and adjust tee accordingly).